### PR TITLE
Add types to router.pp

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: manifests/router.pp [hunk 1]
```diff
@@ -21,16 +21,16 @@
...
-  $mode                    = $qpid::router::params::router_mode,
-  $config_file             = $qpid::router::params::config_file,
-  $router_packages         = $qpid::router::params::router_packages,
-  $open_file_limit         = $qpid::router::params::open_file_limit,
-  $worker_threads          = $qpid::router::params::worker_threads,
-  $hello_interval          = $qpid::router::params::hello_interval,
-  $hello_max_age           = $qpid::router::params::hello_max_age,
-  $service_ensure          = true,
-  $service_enable          = undef,
+  Stdlib::Host $router_id = $qpid::router::params::router_id,
```
Comment (ekohl) [inline manifests/router.pp:24]: I don't know how this is limited. Can it be a FQDN or an IP or just any string?

File: templates/router/header.conf.erb [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 router {
     id: <%= scope['qpid::router::router_id'] %>
-    mode: <%= scope['qpid::router::router_mode'] %>
+    mode: <%= scope['qpid::router::mode'] %>
```
Comment (ekohl) [inline templates/router/header.conf.erb:25]: This looks like an unrelated bug fix. For the changelog it may be good to have it separate. In case we want to cherry pick it.

File: manifests/router.pp (hunk 1 above)
Comment (ehelms) [reply manifests/router.pp:24]: > The unique identifier of the router. This ID is used as the container-id (container name) at the AMQP protocol level. If it is not specified, the router shall generate a random identifier at startup.

That reads like it could be any ole string.

File: templates/router/header.conf.erb (hunk 2 above)
Comment (ehelms) [reply templates/router/header.conf.erb:25]: I can extract it. This change unearthed the bug since `router_mode` got removed from params.pp this removed the parameter and tests pointed out the mismatch.

Comment (ehelms) [reply templates/router/header.conf.erb:25]: https://github.com/theforeman/puppet-qpid/pull/153

**Comment by ehelms** [comment]
Will need a rebase on top of https://github.com/theforeman/puppet-qpid/pull/153
